src/data.table.h

```diff
@@ -98,6 +98,10 @@ extern SEXP sym_index;
 extern SEXP sym_BY;
 extern SEXP sym_starts, char_starts;
 extern SEXP sym_maxgrpn;
+extern SEXP sym_anyna;
+extern SEXP sym_anyinfnan;
+extern SEXP sym_anynotascii;
+extern SEXP sym_anynotutf8;
```

Comment:
I feel we could fill up follow up issues for situation where those metadata could be utilized for further speed ups.

Reply:
Let's use #2879 for that? Since it won't be closed by this PR.